pkg/kots/lint.go

```diff
@@ -344,6 +360,106 @@ func lintWithKubevalSchema(renderedFiles SpecFiles, originalFiles SpecFiles, sch
 	return lintExpressions, nil
 }
 
+func lintTargetMinVersions(specFiles SpecFiles) ([]LintExpression, error) {
+	lintExpressions := []LintExpression{}
+	// separate multi docs because the manifest can be a part of a multi doc yaml file
+	separatedSpecFiles, err := specFiles.separate()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to separate multi docs")
+	}
+
+	url := "http://api.github.com/repos/replicatedhq/kots/commits/%s"
+	token := os.Getenv("GITHUB_API_TOKEN")
+	var bearer = "Bearer " + token
+
+	for _, spec := range separatedSpecFiles {
+		var tv, mv string
+		var tvExists, mvExists bool
+		doc := map[string]interface{}{}
+		if err := goyaml.Unmarshal([]byte(spec.Content), &doc); err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal spec content")
+		}
+		if doc["apiVersion"] == "kots.io/v1beta1" && doc["kind"] == "Application" {
+			if spec, ok := doc["spec"].(map[interface{}]interface{}); ok {
+				tv, tvExists = spec["targetKotsVersion"].(string)
+				mv, mvExists = spec["minKotsVersion"].(string)
+			}
+		}
+
+		// if no min nor target kots version exists, continue to next file
+		if !mvExists && !tvExists {
+			continue
+		}
+
+		lintExpression := LintExpression{
+			Rule:    "invalid-target-kots-version",
+			Type:    "error",
+			Path:    spec.Path,
+			Message: "Invalid targetKotsVersion provided",
+		}
+
+		if tvExists {
+			version := tv
+			if !strings.HasPrefix(version, "v") {
```

Comment:
the code from here on is basically the same, any reason it's not in a separate function and used for both target and min kots verisons?

Reply:
ofcourse, done